dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestUUIDEntityIDServlet.java

```diff
@@ -437,7 +437,52 @@ public void testBasicUUIDIdentity_UUID_Generator_JTA() {
     }
 
     /**
-     * Verify that an entity using a UUID type for its identity can use the UUID generator for primary key generation. XML Variant.
+     * Verify that an entity using a UUID type for its identity can use the UUID generator for primary key generation. Merge
+     */
+    @Test
+    public void testBasicUUIDIdentity_UUID_Generator_Merge_JTA() {
+        EntityManager tem = null;
+
+        try {
+            tem = emfJta.createEntityManager();
+            Assert.assertNotNull(tem);
+            Assert.assertTrue(tem.isOpen());
+
+            UUIDUUIDGenEntity entity = new UUIDUUIDGenEntity();
+            entity.setStrData("Some string data");
+
+            tx.begin();
+            tem.joinTransaction();
+            tem.merge(entity);
```

Comment:
When using "merge", it would be more correct to use the return value, so please update this to:
entity = tem.merge(entity);